File: output/llama_4_maverick/avatar/direct_translation/python/java/failed_translation/codeforces_617_B.java

import java.util.Scanner;

public class codeforces_617_B {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        scanner.nextLine();
        String a = scanner.nextLine().trim();
        String[] parts = a.replaceAll("^0+|0+$", "").split("1");
        long result = 1;
        boolean hasOne = false;
        for (String part : parts) {
            if (part.length() > 0) {
                hasOne = true;
                result *= (part.length() + 1);
            }
        }
        System.out.println(hasOne ? result : 0);
    }
}

File: output/llama_4_maverick/avatar/algo_based_translation/java/python/failed_translation/codeforces_617_B.py
n = int(input())
arr = []
for i in range(n):
    if int(input().split()[0]) == 1:
        arr.append(i)
if not arr:
    print(0)
else:
    result = 1
    for i in range(1, len(arr)):
        result *= arr[i] - arr[i - 1]
    print(result)
